system/pages/admin/news.php

```diff
@@ -0,0 +1,258 @@
+			$p_title = $news['title'];
+			$body = $news['body'];
+			$comments = $news['comments'];
+			$type = $news['type'];
+			$category = $news['category'];
+			$player_id = $news['player_id'];
+			$article_text = $news['article_text'];
+			$article_image = $news['article_image'];
+		}
+		else {
+			if(News::update($id, $p_title, $body, $type, $category, $player_id, $forum_section, $article_text, $article_image, $errors)) {
+				// update forum thread if exists
+				if(isset($forum_section) && Validator::number($forum_section)) {
+					$db->query("UPDATE `" . TABLE_PREFIX . "forum` SET `author_guid` = ".(int) $player_id.", `post_text` = ".$db->quote($body).", `post_topic` = ".$db->quote($p_title).", `edit_date` = " . time() . " WHERE `id` = " . $db->quote($forum_section));
+				}
+
+				$action = $p_title = $body = $comments = $article_text = $article_image = '';
+				$type = $category = $player_id = 0;
+
+				success("Updated successful.");
+			}
+		}
+	}
+	else if($action == 'hide') {
+		News::toggleHidden($id, $errors);
+	}
+
+	if(!empty($errors))
+		error(implode(", ", $errors));
+}
+
+$categories = array();
+foreach($db->query('SELECT `id`, `name`, `icon_id` FROM `' . TABLE_PREFIX . 'news_categories` WHERE `hidden` != 1') as $cat)
+{
+	$categories[$cat['id']] = array(
+		'name' => $cat['name'],
+		'icon_id' => $cat['icon_id']
+	);
+}
+
+if($action == 'edit') {
+	$player = new OTS_Player();
+	$player->load($player_id);
+}
+
+$account_players = $account_logged->getPlayersList();
+$account_players->orderBy('group_id', POT::ORDER_DESC);
+$twig->display('admin.news.form.html.twig', array(
+	'action' => $action,
+	'news_link' => getLink(PAGE),
+	'news_link_form' => '?p=news&action=' . ($action == 'edit' ? 'edit' : 'add'),
+	'news_id' => isset($id) ? $id : null,
+	'title' => isset($p_title) ? $p_title : '',
+	'body' => isset($body) ? htmlentities($body, ENT_COMPAT, 'UTF-8') : '',
+	'type' => isset($type) ? $type : null,
+	'player' => isset($player) && $player->isLoaded() ? $player : null,
+	'player_id' => isset($player_id) ? $player_id : null,
+	'account_players' => $account_players,
+	'category' => isset($category) ? $category : 0,
+	'categories' => $categories,
+	'forum_boards' => getForumBoards(),
+	'forum_section' => isset($forum_section) ? $forum_section : null,
+	'comments' => isset($comments) ? $comments : null,
+	'article_text' => isset($article_text) ? $article_text : null,
+	'article_image' => isset($article_image) ? $article_image : null
+));
+
+$query = $db->query('SELECT * FROM ' . $db->tableName(TABLE_PREFIX . 'news'));
+$newses = $tickers = $articles = array();
+foreach ($query as $_news) {
+	$_player = new OTS_Player();
+	$_player->load($_news['player_id']);	
+
+	if($_news['type'] == constant('NEWS')){
+		$newses[] = array(
+			'id' => $_news['id'],
+			'archive_link' => getLink('news') . '/archive/' . $_news['id'],
+			'title' => $_news['title'],
+			'date' => $_news['date'],
+			'player_name' => $_player->getName(),
```

Comment:
It may happen that the player cannot be found, because it was deleted or any other reason.

In this case an exception: " Fatal error: Uncaught E_OTS_NotLoaded" will be thrown when callin getName().

You need to check if player exist before calling this.